hnn_core/network.py

```diff
@@ -251,8 +253,15 @@ def _add_external_feeds(self):
         unique_keys = sorted(self.p_unique.keys())
         self.feedname_list += unique_keys
 
-    def _create_gid_dict(self):
-        """Creates gid dicts."""
+        self._update_gid_dict()  # add feeds -> update the gid_dict
+
+    def _update_gid_dict(self):
+        """Creates gid dict from scratch every time called.
+
+        Any method that adds real or artificial cells to the network must
+        call this to update the list of GIDs. Note that it's based on the
+        content of pos_dict and the lists of real and artificial cell names.
+        """
```

Comment:
This is a great idea @cjayb and will allow the user to more flexibly create or destroy cells or feeds! Since `Network.gid_dict` and `Network.pos_dict` are so intimately tied, would it make sense to combine their assignment under a unified method?

Reply:
Just to clarify, this will only add or remove keys, not modify the values, right? E.g., if I have `L2_Pyramidal: range(0, 100)` and `L5_Pyramidal: range(100, 200)`, do you imagine each of the key-value pairs to be updated when new cells are added?